api/controllers/Irc_EntryController.js

```diff
@@ -79,16 +82,12 @@ module.exports = {
       detainee: detainee
     }),
 
-  saveDetainee: function (request_body) {
-    var person_id = this.getPID(request_body);
-    return Detainees.findOrCreate({
-      id: person_id
-    }, {
-      id: person_id,
-      cid_id: request_body.cid_id,
-      gender: request_body.gender,
-      centre: request_body.centre
-    });
+  saveDetainee: function (detainee) {
+    return Detainees.findOrCreate({id: detainee.id}, detainee);
+  },
+
+  updateDetainee: function (detainee) {
+    return Detainees.update({id: detainee.id}, detainee);
```

Comment:
this doesn't handle out of order updates, you should be comparing a timestamp before persisting the data in the payload